File: jdr-final/src/main/java/com/monapp/entity/Personnage.java
package com.monapp.entity;


import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Personnage {

	// CREATION DES VARIABLES :
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@JsonView(Views.Common.class)
	private int id;
	
	@Column
	@JsonView(Views.Common.class)
	private String nom;
	
	@Column
	@JsonView(Views.Common.class)
	private String job;
	
	@Column
	@JsonView(Views.Common.class)
	private int positionActuelle;
	
	@OneToOne (fetch=FetchType.EAGER)
	@JoinColumn(name ="inventaire")
	@JsonView(Views.Utilisateur.class)
	private Inventaire inventaire;
	
	@ManyToOne (fetch=FetchType.EAGER)
	@JoinColumn(name ="utilisateur")
	@JsonView(Views.Utilisateur.class)
	private Utilisateur utilisateur;
	
	//CONSTRUCTEUR
	
	private Personnage() {}
	
	private Personnage(String nom, String job, int positionActuelle) {
		this.nom = nom;
		this.job = job;
		this.positionActuelle = positionActuelle;
	}

	

	
}

File: app/personnage.ts
import {Utilisateur} from './utilisateur';
import {Inventaire} from './inventaire';

export class Personnage {

 constructor ( private _id: number,
  private _nom: string,
  private _job: string,
  private _positionActuelle: number,
  private _inventaire: Inventaire,
  private _utilisateur: Utilisateur ){}


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get nom(): string {
    return this._nom;
  }

  set nom(value: string) {
    this._nom = value;
  }

  get job(): string {
    return this._job;
  }

  set job(value: string) {
    this._job = value;
  }

  get positionActuelle(): number {
    return this._positionActuelle;
  }

  set positionActuelle(value: number) {
    this._positionActuelle = value;
  }

  get inventaire(): Inventaire {
    return this._inventaire;
  }

  set inventaire(value: Inventaire) {
    this._inventaire = value;
  }

  get utilisateur(): Utilisateur {
    return this._utilisateur;
  }

  set utilisateur(value: Utilisateur) {
    this._utilisateur = value;
  }
}
